custom_components/tahoma/climate_st.py

```diff
@@ -99,6 +102,16 @@ def __init__(
     async def async_added_to_hass(self):
         """Register temperature sensor after added to hass."""
         await super().async_added_to_hass()
+
+        # The Somfy Thermostat requires a temperature sensor
+        base_url = self.device.deviceurl.split("#", 1)[0]
+        entity_registry = await self.hass.helpers.entity_registry.async_get_registry()
+        self._temp_sensor_entity_id = next(
+            entity_id
```

Comment:
Nice change!! I just wanted to respond that the other notation was not really clear haha. :-)

What is the result when next() can not find a sensor? Will it be None? Otherwise, do we need to add a `None` fallback. https://stackoverflow.com/questions/8653516/python-list-of-dictionaries-search

Reply:
"If the default parameter is omitted and the iterator is exhausted, it raises StopIteration exception."

I guess `None` is ok as a fallback.

Reply:
If you set None as default value you will have to protect the code each time the sensor is used. Normally it should not happen but I feel the exception will be more useful.

Reply:
you're right, as is, with None as a default, next line will fail with:
```python

2020-08-05 12:22:06 ERROR (MainThread) [homeassistant.components.climate] Error while setting up tahoma platform for climate
Traceback (most recent call last):
  File "/Users/vlb/PycharmProjects/ha-tahoma/venv/lib/python3.8/site-packages/homeassistant/helpers/entity_platform.py", line 187, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/Users/vlb/PycharmProjects/ha-tahoma/venv/lib/python3.8/site-packages/homeassistant/helpers/entity_platform.py", line 293, in async_add_entities
    await asyncio.gather(*tasks)
  File "/Users/vlb/PycharmProjects/ha-tahoma/venv/lib/python3.8/site-packages/homeassistant/helpers/entity_platform.py", line 450, in _async_add_entity
    await entity.async_added_to_hass()
  File "/Users/vlb/PycharmProjects/ha-tahoma/Config/custom_components/tahoma/climate_st.py", line 118, in async_added_to_hass
    async_track_state_change(
  File "/Users/vlb/PycharmProjects/ha-tahoma/venv/lib/python3.8/site-packages/homeassistant/helpers/event.py", line 92, in async_track_state_change
    entity_ids = tuple(entity_id.lower() for entity_id in entity_ids)
TypeError: 'NoneType' object is not iterable
```

One solution is to only track change line 118 if sensor isn't None, or to fail wit the StopIteration exception.

Reply:
second solution is better to me, it allows the component to start gracefully and shows the climate device in the UI, just without the current temperature.
![image](https://user-images.githubusercontent.com/5701372/89402496-416b6b80-d717-11ea-9bd1-28570206b4bc.png)

Reply:
Would it make sense to raise a custom warning for the logging, which makes it easier to debug?

Reply:
sure, that's ok with me

Reply:
I don't know anything about climate devices... But I would opt for a way where we start gracefully indeed and just skip the temperature. Maybe add a warning to the log that it could not find the temperature device.

Reply:
However, this device always has a temperature sensor I believe? Thus we are looking at a super edge case at the moment. Not sure if we need to spend a lot of work on it.

Reply:
Indeed, it's part of the physical device, so unless they change the url of the device, I don't see how it could not be found...

Reply:
this would start gracefully in such a case:
```py
if self._temp_sensor_entity_id:
    async_track_state_change(
        self.hass, self._temp_sensor_entity_id, self._async_temp_sensor_changed
    )
else:
    _LOGGER.warning("Temperature sensor could not be found for entity %s", self.name)
```